### PR TITLE
feat: store sort key in NG metadata

### PR DESCRIPTION
**Pull request by crepererum**
The sort key is optional and currently only produced by `iox_tests`.
Writing it within the ingester/compactor is tracked by #3968. The sort
key is read by the querier (and this will be verified by the query tests
and is required to merge #4103).

### CODE REVIEW

File: iox_tests/src/util.rs [hunk 1]
```diff
@@ -435,3 +443,60 @@ impl TestTombstone {
 pub fn now() -> Time {
     Time::from_timestamp(0, 0)
 }
+
+/// Sort mutable batch into arrow record batch and sort key.
+fn sort_mutable_batch(batch: MutableBatch) -> (RecordBatch, SortKey) {
+    // build dummy sort key
```
Comment (alamb) [inline iox_tests/src/util.rs:449]: I know this is just for testing, but I think you might be able to make this a lot simpler by using a datafusion plan.

Something like (untested)

```rust
    let record_batch = batch.to_arrow(Selection::All).unwrap();
    let mem_table = MemTable::new(record_batch.schema(), vec![vec![record_batch]});
    let mut ctx = SessionContext::new();
    let sorted_results = ctx.register_table(mem_table)
      .sort(exprs)
      .collect()
```

I am happy to help with a follow on PR if needed

(making a API in datafusion to scan a set of batches would be great too 🤔 )

File: querier/src/chunk/mod.rs [hunk 2]
```diff
@@ -310,10 +309,11 @@ mod tests {
             catalog.time_provider(),
         );
 
+        // set up catalog
         let lp = vec![
-            "table,tag1=WA field_int=1000 8000",
-            "table,tag1=VT field_int=10 10000",
-            "table,tag1=UT field_int=70 20000",
+            "table,tag1=WA field_int=1000i 8000",
```
Comment (alamb) [inline querier/src/chunk/mod.rs:314]: why this change?

**Review by alamb — APPROVED** [review]

File: querier/src/chunk/mod.rs (hunk 2 above)
Comment (crepererum) [reply querier/src/chunk/mod.rs:314]: Because the test helpers (`iox_tests`) now sort the data before storing it, so the "sort key" isn't a lie (that's the code path you commented on above). And we don't run a full datafusion query here, it's just a simple "scan" operation, so "orderIn = orderOut".

File: iox_tests/src/util.rs (hunk 1 above)
Comment (crepererum) [reply iox_tests/src/util.rs:449]: I leave that to you then. TBH just the simple sort operation for testing was way more complicated than I wanted it to be, but as expressed below I didn't want the sort key to be a lie, so on the long run doing it proper is worth it.

Comment (NGA-TRAN) [reply iox_tests/src/util.rs:449]: But eventually, we need that `compute_sort_key` to be smart because we want all files sort on the same sort key. I think that is a bigger task listed in #3968
